### PR TITLE
Npshenichnyy/sds filebased integration tests (WIP)

### PR DESCRIPTION
**Pull request by nikolay-pshenichny**
WIP

### CODE REVIEW

**Comment by nikolay-pshenichny** [comment]
@ramaraochavali @shakti-das , fyi

File: tests/integration/security/file_mounted_certs/main_test.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+values:
+  global:
+    logging:
+      level: "default:debug"
+    proxy:
+      logLevel: debug
+      componentLogLevel: "misc:debug"
+  pilot:
+    env:
+      PILOT_SKIP_VALIDATE_TRUST_DOMAIN: "true"
```
Comment (shakti-das) [inline tests/integration/security/file_mounted_certs/main_test.go:140]: We set a bunch of other env values for Pilot, ex: DNS_ADDR, ENABLE_CA_SERVER, PILOT_ENABLE_PROTOCOL_SNIFFING_*. Are they being set somewhere else?

Comment (nikolay-pshenichny) [reply tests/integration/security/file_mounted_certs/main_test.go:140]: @shakti-das , I am actually not sure if those are set. will double-check. Thank you!

Comment (nikolay-pshenichny) [reply tests/integration/security/file_mounted_certs/main_test.go:140]: The following env vars are present in the istiod pod:
```
    - name: REVISION
      value: default
    - name: JWT_POLICY
      value: first-party-jwt
    - name: PILOT_CERT_PROVIDER
      value: istiod
    - name: POD_NAME
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: metadata.name
    - name: POD_NAMESPACE
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: metadata.namespace
    - name: SERVICE_ACCOUNT
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: spec.serviceAccountName
    - name: KUBECONFIG
      value: /var/run/secrets/remote/config
    - name: PILOT_SKIP_VALIDATE_TRUST_DOMAIN
      value: "true"
    - name: PILOT_TRACE_SAMPLING
      value: "1"
    - name: PILOT_ENABLE_PROTOCOL_SNIFFING_FOR_OUTBOUND
      value: "true"
    - name: PILOT_ENABLE_PROTOCOL_SNIFFING_FOR_INBOUND
      value: "true"
    - name: INJECTION_WEBHOOK_CONFIG_NAME
      value: istio-sidecar-injector
    - name: ISTIOD_ADDR
      value: istiod.istio-system.svc:15012
    - name: PILOT_ENABLE_ANALYSIS
      value: "false"
    - name: CLUSTER_ID
      value: Kubernetes
    - name: CENTRAL_ISTIOD
      value: "false"
```

File: tests/integration/security/file_mounted_certs/p2p_mtls_test.go [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+
+	configDump, err := clientWorkloads[0].Sidecar().Config()
+	if err != nil {
+		t.Fatalf("Unable to retrieve config_dump from the client. %v", err)
+	}
+
+	clusterName := clusterName(server, echo.Port{ServicePort: 8443})
+
+	validator := structpath.ForProto(configDump)
+	destCluster := validator.Select("{.configs[*].dynamicActiveClusters[?(@.cluster.name == '%s')]}", clusterName)
```
Comment (ramaraochavali) [inline tests/integration/security/file_mounted_certs/p2p_mtls_test.go:304]: This validation of config is not needed. We do have unit tests that validate this behaviour. We just want e->e verification here.

File: tests/integration/security/file_mounted_certs/main_test.go (hunk 1 above)
Comment (shakti-das) [reply tests/integration/security/file_mounted_certs/main_test.go:140]: We should set `ENABLE_CA_SERVER` as false (its default is true) and `DNS_ADDR` as "" (its default is ":15053").

File: tests/integration/security/file_mounted_certs/p2p_mtls_test.go (hunk 2 above)
Comment (nikolay-pshenichny) [reply tests/integration/security/file_mounted_certs/p2p_mtls_test.go:304]: Wanted to have this just to be on the safe side. ie - to know for sure that filebased certs were configured/used

**Pull request closed without merging**